src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/UserService.kt

```diff
@@ -1,20 +1,41 @@
 package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
 
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.manageusers.AuthSource
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.manageusers.ManageUsersClient
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.common.isEmail
-import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.ContactDetails
 
 @Service
 class UserService(private val manageUsersClient: ManageUsersClient) {
 
-  fun getContactDetails(username: String): ContactDetails? {
-    val mayBeUserDetails = manageUsersClient.getUsersDetails(username) ?: return null
+  fun getUser(username: String): User? {
+    val userDetails = manageUsersClient.getUsersDetails(username) ?: return null
 
-    return if (username.isEmail()) {
-      ContactDetails(name = mayBeUserDetails.name, email = username)
-    } else {
-      return ContactDetails(name = mayBeUserDetails.name, email = manageUsersClient.getUsersEmail(username)?.email)
-    }
+    return User(
+      username = username,
+      userType = if (userDetails.authSource == "nomis") UserType.PRISON else UserType.AUTH,
+      name = userDetails.name,
+      email = if (username.isEmail()) username else manageUsersClient.getUsersEmail(username)?.email,
+    )
   }
+
+  fun getServiceAsUser(): User {
```

Comment:
This could be be moved into a companion method in the service so ...

```
companion object {
  fun getServiceAsUser() = User(
      username = ServiceName.BOOK_A_VIDEO_LINK_SERVICE.name,
      userType = UserType.SERVICE,
      name = ServiceName.BOOK_A_VIDEO_LINK_SERVICE.name,
    )
}
```

This then removes the need to to autowire into other components e.g. `UserService.getServiceAsUser()`